plugins/net_plugin/net_plugin.cpp

```diff
@@ -2056,6 +2056,9 @@ namespace eosio {
                   ("ep", c->peer_name())("lib", msg.last_irreversible_block_num)("head", msg.head_num)
                   ("id", msg.head_id.str().substr(8,16)) );
          c->syncing = false;
+         if (c->sent_handshake_count > 0) {
```

Comment:
because of this we will send it twice then:
```
void connection::handle_message( const handshake_message& msg ) {
...
      if (msg.generation == 1) {
      ...
         if( sent_handshake_count == 0 ) {
            send_handshake();
         }
      }
      ...
      my_impl->sync_master->recv_handshake( shared_from_this(), msg );

}
```
As an alternative, since we process `recv_handshake` anyway, we can remove it from `handle_message` and make unconditional inside `recv_handshake`.
What do you think?